packages/zoe/src/internal-types.js

```diff
@@ -36,56 +33,174 @@
  */
 
 /**
- * @template OC - the offer outcome
- * @typedef {Object} InviteHandler<OC>
- * @property {OfferHook<OC>} invoke
+ * @template T
+ * @typedef {Object} Table
+ * @property {(record: any) => record is T} validate
+ * @property {<H>(record: Omit<T, 'handle'>, handle: H) => H} create
```

Comment:
The `handle` parameter to `create` is optional.
```suggestion
 * @property {<H>(record: Omit<T, 'handle'>, handle?: H) => H} create
```

I don't understand why you only parameterize this one method with `H` rather than
   * parameterizing the table as a whole with `H`, and
   * using `H` rather than `any` as the type of the `handle` parameters.

tables.js itself has a `Handle<H>` which also seems to about typing these handles.

Reply:
Btw, `Omit` is really cool! It does make me worried about how complex the TypeScript type system has gotten. How many other obscure bells and whistles like this are there?